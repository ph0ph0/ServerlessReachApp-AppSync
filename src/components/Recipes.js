import React, { useState } from 'react'
import {useQuery, useSubscription} from '@apollo/react-hooks'
import ListRecipes from '../graphql/queries/ListRecipes'
import NewRecipeSubscription from '../graphql/subscriptions/NewRecipeSubscription'
import { css } from 'glamor'

const Recipes = () => {

    const [recipesState, setRecipeState] = useState([])

    const { data, loading, error} = useQuery(ListRecipes)

    console.log(`datas: ${JSON.stringify(data)}`)

    //Will need some sort of state to 
    //hold the data that comes from the query. Then, everytime the 
    //subscription fires, we append it to the state.

    //We want to do the first fetch when the component mounts and then update thereafter. Actually, 
    //useQuery fires when the component first mounts and so we dont need to put this into the
    //useEffect statement. We also need to provide an unmount listener call

    return (
        <h1>Recipes!</h1>
    )
   }

const styles = {
    title: {

    },
    subtitle: {

    },
    recipe: {

    },
    container: {

    }
}

export default Recipes