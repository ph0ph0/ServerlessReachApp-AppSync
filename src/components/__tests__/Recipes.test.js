import React from 'react'
import Recipes from '../Recipes'
import {render, cleanup } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import { MockedProvider } from '@apollo/react-testing'
import ListRecipes from '../../graphql/queries/ListRecipes'

afterEach(cleanup)

it("renders", () => {
    const { asFragment } = render(<Recipes />)
    expect(asFragment()).toMatchSnapshot()
})

const mocks = [
    {
        request: {
            query: ListRecipes
        }, 
        result: {
            data: {
                listRecipes: {
                    
                    name: "Tuna Rolll",
                    id: "1",
                    instructions: ["Mix Tuna"],
                    ingredients: ["Tuna"]
                    
                }
            }
        }
    }
]

it("gets data from ListRecipes", () => {
    const { getByTestId } = render(
    <MockedProvider mocks = {mocks} addTypename = {false} >
        <Recipes />
    </MockedProvider>)
})
