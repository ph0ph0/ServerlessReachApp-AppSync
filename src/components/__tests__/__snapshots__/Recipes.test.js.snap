// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <h1>
    Recipes!
  </h1>
</DocumentFragment>
`;
